traffic_ops/traffic_ops_golang/server/servers.go

```diff
@@ -856,6 +856,12 @@ func getServers(h http.Header, params map[string]string, tx *sqlx.Tx, user *auth
 `
 	}
 
+	if version.Major >= 5 {
+		if params["type"] == "RASCAL" {
+			params["type"] = "TRAFFIC_MONITOR"
+		}
+	}
+
```

Comment:
We shouldn't do this; you can still make server types, so someone could make their own RASCAL type and then never be able to filter on that.

Reply:
I see your point but how do I maintain backward compatibility in terms of T3C?

This change was done to ensure T3c clients and TS that monitors servers and ask for servers with type rascal, TO sends the servers with type TM.

Reply:
That's a hard question, but the other thing is that t3c doesn't use APIv5, it uses v4 and v3. So old versions of t3c wouldn't be kept compatible with this. In a version of t3c that _does_ use APIv5, we could make it just query for `TRAFFIC_MONITOR` instead, but that doesn't help backwards compatibility.

Maybe the best thing to do would be to make it return *both* `TRAFFIC_MONITOR` *and* `RASCAL`? That makes things a bit more annoying for operators and clients, potentially, but it makes it so there's no type you can't filter for, at least as long as you use your types responsibly.

Reply:
So, this change works, right? Since for v5 it is where I am returning `TRAFFIC_MONITOR`

Reply:
Well what I'm saying is that since they don't use v5 yet, they can just change the parameter they send to be TRAFFIC_MONITOR instead of RASCAL at that point.